packages/react/src/dialog/portal/DialogPortalContext.ts

```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+
+export const DialogPortalContext = React.createContext(false);
+
+export function useThrowIfPortalIsMissing() {
+  const value = React.useContext(DialogPortalContext);
+  if (!value) {
+    throw new Error('Base UI: <Dialog.Portal> is missing.');
+  }
+}
```

Comment:
Throws inside each component's `Popup` part if the `Portal` part is missing as a parent. 

Note: a `Dialog` inside a `Dialog` for instance, with the nested one missing a portal, won't throw, but that setup doesn't have bugs